include/time_series/interface.hpp

```diff
@@ -0,0 +1,107 @@
+/**
+ * @file interface.hpp
+ * @author Vincent Berenz
+ * license License BSD-3-Clause
+ * @copyright Copyright (c) 2019, Max Planck Gesellshaft.
+ */
+
+#pragma once
+
+#include <cstddef>
+#include <limits>
+
+namespace time_series
+{
+typedef long int Index;
+typedef long double Timestamp;
+
+/**
+ * \brief Interface for time series.
+ * A time_series implements  \f$ X_{{oldest}:{newest}} \f$ which can
+ * safely be accessed from either multiple threads or multiple processes.
+ *
+ * this object has the following properties:
+ * - an oldest timeindex \f$ oldest\f$,
+ * - a newest timeindex \f$ newest \f$,
+ * - a value \f$ X_i \f$ for each  \f$ i \in \{oldest, oldest + 1 , ...,
+ * newest\} \f$,
+ * - a length \f$length\f$
+ * - and a maximum length \f$maxlength\f$
+ */
+template <typename T>
+class TimeSeriesInterface
+{
+public:
+    /*! \brief returns \f$ newest \f$. waits if the time_series is empty.
+     */
+    virtual Index newest_timeindex() = 0;
+
+    /*! \brief returns the number of element contained in the queue.
+     */
+    virtual Index count_appended_elements() = 0;
```

Comment:
How is this different from `length`?

Reply:
I saw later that this is the number of elements appended since start (not the number of elements currently in the queue).  Maybe the comment can be clarified.

Reply:
indeed  +1